src/gamestates/cMainMenuGameState.cpp

```diff
@@ -50,10 +41,17 @@ cMainMenuGameState::cMainMenuGameState(cGame &theGame) : cGameState(theGame), te
 
     int creditsX = (game.m_screenW / 2) - buttonWidth;
     const cRectangle &creditsRect = cRectangle(creditsX, 0, buttonWidth, buttonHeight);
-    gui_btn_credits = new cGuiButton(textDrawer, creditsRect, "CREDITS", buttonKinds);
-    gui_btn_credits->setTextAlignHorizontal(buttonTextAlignment);
-    gui_btn_credits->setOnLeftMouseButtonClickedAction(new cGuiActionToGameState(GAME_CREDITS, true));
 
+    gui_btn_credits = GuiButtonBuilder()
```

Comment:
I *love* the builder pattern. Nice to see it here!

Reply:
It was a goal of this branch !